cdap-watchdog/src/main/java/co/cask/cdap/logging/write/FileMetaDataManager.java

```diff
@@ -184,4 +220,135 @@ public Integer apply(DatasetContext<Table> ctx) throws Exception {
   public interface DeleteCallback {
     public void handle(Location location);
   }
+
+  /**
+   * Upgrades the log meta table
+   * Note: Currently this supports upgrade from 2.6 to 2.8 and does upgrade for namepspaces
+   *
+   * @throws Exception
+   */
+  public void upgrade(final DatasetFramework dsFramework) throws Exception {
+    final Transactional<DatasetContext<Table>, Table> oldLogMDS;
+    // get the old log meta data store from the default namespace
+    oldLogMDS = Transactional.of(txExecutorFactory, new Supplier<DatasetContext<Table>>() {
+      @Override
+      public DatasetContext<Table> get() {
+        try {
+          Table table = DatasetsUtil.getOrCreateDataset(dsFramework, Id.DatasetInstance.from
+                                                          (Constants.DEFAULT_NAMESPACE_ID, Joiner.on(".").join(
+                                                            Constants.SYSTEM_NAMESPACE,
+                                                            LoggingConfiguration.LOG_META_DATA_TABLE)),
+                                                        "table", DatasetProperties.EMPTY,
+                                                        DatasetDefinition.NO_ARGUMENTS, null);
+          return DatasetContext.of(table);
+        } catch (Exception e) {
+          LOG.error("Failed to access {} table", Joiner.on(".").join(Constants.SYSTEM_NAMESPACE,
```

Comment:
no need to catch, log, then wrap in a runtime exception.  The upgrade method already throws an exception, just let the caller handle it.